app/assets/stylesheets/common/base/d-image-grid.scss

```diff
@@ -0,0 +1,52 @@
+.d-image-grid:not([data-disabled]) {
+  $grid-column-gap: 6px;
+
+  &[data-columns] {
+    display: flex;
+    flex-wrap: wrap;
+  }
+
+  &[data-columns="2"] > * {
+    flex-basis: calc(50% - ($grid-column-gap / 2));
+    margin-right: $grid-column-gap;
+  }
+
+  &[data-columns="3"] > * {
+    flex-basis: calc(33.33% - ($grid-column-gap * 0.667));
+    margin-right: $grid-column-gap;
+  }
+
+  .d-image-grid-column {
+    box-sizing: border-box;
+
+    &:last-child {
+      margin-right: 0;
+    }
+
+    > img {
+      margin-bottom: $grid-column-gap;
+    }
```

Comment:
I think we should force the images to fill the width of the column. In my local test topic, my images were previously set to very small dimensions in the markdown, which made the masonry layout look very strange

<img width="404" alt="SCR-20230605-opcj" src="https://github.com/discourse/discourse/assets/6270921/a6514017-c694-47d7-9816-cc53afb4e1ee">

Similarly, when using the mobile mode, but with a wide viewport (i.e. a device in landscape mode), the images show up weirdly wrapped in two separate lists. I think this would also be solved by a `width: 100%`

<img width="1447" alt="SCR-20230605-oppn" src="https://github.com/discourse/discourse/assets/6270921/4bab3d09-e8e9-4c8f-bd30-1a8d5fb809d4">

But also... we might need to reconsider the landscape-mobile experience here. Maybe we need a max-width on the grid? I guess ideally we'd dynamically change the number of columns when someone rotates a device... but that would be quite tricky to achieve.

Reply:
Yeah, this is indeed a "small images" issue (I can't repro with unmodified image sizes on mobile portrait or landscape). No big objections to forcing a `width: 100%`.